rfcs/convergence/exposing-popper-options.md

```diff
@@ -0,0 +1,203 @@
+result would be that a popper configuration is exposed for one component but not others.
+
+## Detailed Design or Proposal
+
+### Equal popper configuration
+
+Each converged component that uses popper should allow an equal level of configuration. This ensures that our positioned
+components can behave consistently across the library. Any fixes that require exposing another popper configuration will
+be available to **all** components that use popper.
+
+This proposal does not mean that all popper options must be public, but that if one component exposes a popper configuration,
+then all components should do so equally.
+
+We can achieve this with an agreement that the current `PositioningProps` would be supported by all v9 components, while
+internal popper options can still belong to `PopperOptions` type in the `react-positioning` package.
+
+```tsx
+import { PositioningProps } from '@fluentui/react-positioning';
+
+export interface TooltipProps extends PositioningProps {}
+
+export interface MenuProps extends PositioningProps {}
+
+export interface PopoverProps extends PositioningProps {}
+```
+
+### Shorhand `placement` configuration
+
+The next step to this proposal is to make sure that all components which use a positioned element should allow users
+to configure popper through a `placement` prop that is behaves like a shorthand. The usage examples below will detail
+the usage of this shorthand.
+
+```tsx
+import { PositioningProps } from '@fluentui/react-positioning';
+
+export interface TooltipProps {
+  placement?: PositioningProps;
+}
+
+export interface MenuProps {
+  placement?: PositioningProps;
+}
+
+export interface PopoverProps {
+  placement?: PositioningProps;
+}
+```
+
+Example usage:
+
+```tsx
+type PlacementShorthand =
+  | 'above'
+  | 'above-start'
+  | 'above-end'
+  | 'below'
+  | 'below-start'
+  | 'below-end'
+  | 'before'
+  | 'before-top'
+  | 'before-bottom'
+  | 'after'
+  | 'after-top'
+  | 'after-bottom';
+
+// Shorthand for `position` and `align` options
+<PositionedComponent placement="above-start">
+<PositionedComponent placemant={{position: 'above', align: 'start'}}>
+
+<PositionedComponent placement="above">
+<PositionedComponent placemant={{position: 'above'}}>
+
+// Also allows more complex configuration
+<PositionedComponent placement={{position: 'above', align: 'start', autoSize: true}}>
+```
+
+Slots that use popper should use a similar pattern if a top level `placement` option is not clear enough:
+
+```tsx
+import { PositioningProps } from '@fluentui/react-positioning';
+
+export interface ComponentWithPositionedSlotProps {
+  positionedSlot: React.HTMLAttributes<HTMLElement> & { placement?: PositioningProps };
+}
+```
+
+This way we can safely configure `usePopper` internally with a simple object spread that does not require knowing what
+props are used by the component props:
+
+```tsx
+// � Existing usage
```

Comment:
![image](https://user-images.githubusercontent.com/14183168/128847941-9211851d-39aa-4590-ba2d-dcadac7166af.png)

Is everything okay with this emoji? 🤨